domain/port/service/watcher.go

```diff
@@ -103,17 +105,17 @@ func (s *WatchableService) endpointToMachineMapper(
 		return e.Changed()
 	})
 
-	machineUUIDs, err := s.st.GetMachinesForEndpoints(ctx, endpointUUIDs)
+	machineNames, err := s.st.GetMachineNamesForEndpoints(ctx, endpointUUIDs)
 	if err != nil {
 		return nil, err
 	}
 
-	newEvents := make([]changestream.ChangeEvent, 0, len(machineUUIDs))
-	for _, machineUUID := range machineUUIDs {
+	newEvents := make([]changestream.ChangeEvent, 0, len(machineNames))
+	for _, machineName := range machineNames {
 		newEvents = append(newEvents, changeEventShim{
 			changeType: changestream.Update,
 			namespace:  "machine",
-			changed:    machineUUID,
+			changed:    machineName.String(),
```

Comment:
Didn't you have a seperate PR to change this? Can this one be re-based now that is in?

Reply:
This was the PR I was thinking of, I guess its a different thing though https://github.com/juju/juju/pull/18229

Reply:
Yeah unit vs machine